apis/python/tests/test_tiledb_object.py

```diff
@@ -40,3 +41,65 @@ def test_mutually_exclusive_create_args():
 
     with pytest.raises(TypeError):
         FakeTileDBObject("child", parent=p, context=context)
+
+
+def test_open_close():
+    o = FakeTileDBObject("parent")
+
+    assert not o._open_mode
+    o.open("r")
+    assert o._open_mode == "r"
+    o.close()
+    assert not o._open_mode
+    o.close()  # no-op
+    assert not o._open_mode
+
+    with o.open("w"):
+        assert o._open_mode == "w"
+    assert not o._open_mode
```

Comment:
I think this motivates the creation of a public `closed` (or `is_open`) property on the object, much like a Python file.